src/modules/graph-authentication/graph-authentication.service.test.ts

```diff
@@ -0,0 +1,57 @@
+import { AuthenticationResult, ClientCredentialRequest } from '@azure/msal-node';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+
+import { GraphAuthenticationFailedException } from './exception/graph-authentication-failed.exception';
+import { GraphAuthenticationService } from './graph-authentication.service';
+
+describe('GraphAuthenticationService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const clientId = valueGenerator.string();
+  const clientSecret = valueGenerator.string();
+  const authority = valueGenerator.string();
+  const scope = valueGenerator.string();
+  const accessToken = valueGenerator.string();
+  // The below returns more than just the accessToken, but we only need the accessToken for the test
+  const successfulGraphAcquireTokenByClientCredential = {
+    accessToken,
+  } as AuthenticationResult;
+
+  const graphAuthenticationConfig = {
+    clientId,
+    clientSecret,
+    authority,
+    scope,
+  };
+
+  let graphAuthenticationService: GraphAuthenticationService;
+  let acquireTokenByClientCredentialSpy: jest.SpyInstance<Promise<AuthenticationResult>, [request: ClientCredentialRequest], any>;
+
+  beforeEach(() => {
+    graphAuthenticationService = new GraphAuthenticationService(graphAuthenticationConfig);
+    acquireTokenByClientCredentialSpy = jest.spyOn(graphAuthenticationService['cca'], 'acquireTokenByClientCredential');
```

Comment:
I don't think we should be spying on a `private` field of the class, it makes the test too implementation-dependant. Let me know if you want to have a chat about a different approach